practice/lib/practice/parity_outlier.ex

```diff
@@ -1,38 +1,26 @@
-defmodule Kata do
+defmodule Parity_outlier do
 
   require Integer
 
   def find_outlier(list) do
-    with value_of_list <- Enum.sum(list),
-         boolean_value <- Integer.is_even(value_of_list),
-         list_w_zeros <- parse_atom(list, boolean_value) do
-          filter_list(list_w_zeros)
-          |> List.first
-    end
-  end
+    value = eval_even_or_odd(list)
 
-  def parse_atom(list, boolean) do
-    if boolean == true do
-      Enum.map(list, fn x ->
-        if Integer.is_even(x) do
-          x
-        else
-          "0"
-        end
-      end)
-    else
-      Enum.map(list, fn x ->
-      if Integer.is_odd(x) do
-        x
-      else
-      "0"
-      end
-      end)
-    end
+    list
+    |> parse_even_or_odd(value)
+    |> Enum.sum
   end
 
-  def filter_list(list) do
+  def eval_even_or_odd(list) do
     list
-    |> Enum.reject(fn x -> x == "0" end)
+    |> Enum.map(fn x -> Integer.is_even(x) end)
+    |> Enum.count(fn x -> x == true end)
+  end
```

Comment:
It looks like this function returns the number of even elements in `list`. Is that the intent? If so, it might help to name this function in a way that helps the reader understand what it's for. 

Since `Integer.is_even` returns a boolean, you could also combine two lines into one here by replacing lines 15 and 16 with something like:
```
|> Enum.count(fn x -> Integer.is_even(x) end)
```

Reply:
Great call out. With this change, it looks more cleaner. I also deleted the piping, since it's no needed at this point.